llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/MaterializeEncoding.cpp

```diff
@@ -50,17 +50,20 @@ static RankedTensorType getOriginalTypeWithEncoding(RankedTensorType type) {
                                originalType.getElementType(), encoding);
 }
 
+static RankedTensorType getOriginalTypeWithoutEncoding(RankedTensorType type) {
+  return RankedTensorType::get(type.getShape(), type.getElementType());
+}
```

Comment:
Having these two functions side by side here, named respectively `getOriginalTypeWithEncoding` and `getOriginalTypeWithoutEncoding`, suggests that their results should differ only in the presence of the encoding.

But at the moment, they differ in more than that: the existing `getOriginalTypeWithEncoding` walks up the optional `original_type` attribute. Its purpose is to recover the true original tensor type before it was padded. This is because, as long as `set_encoding` does not have padding semantics like `pack` does, we are forced to separately insert `pad` ops, resulting in padded shapes, so just dropping the encoding afterwards does not fully recover the original un-padded shape. The only thing that does recover the orignal un-padded shape is the `original_type` attribute, if it's present.  (If it's not present, that means it's not needed as there was no type change due to padding).

In any case, the terminology "original type" is already taken, in this context, by this specific concept.

So if you do want that original type, you could change the implementation of this function like:

```c++
static RankedTensorType getOriginalTypeWithoutEncoding(RankedTensorType type) {
  RankedTensorType originalType = getOriginalTypeWithEncoding(type);
  return RankedTensorType::get(originalType.getShape(), originalType.getElementType());
}
```

On the other hand, if you specifically do not want that and really just want to drop the encoding attribute but not recover the un-padded shape, the current code as-is is good, but please rename this function to not say "original type". Maybe "drop encoding" is more to the point.

Reply:
Thanks for the details! What I want is the same type without encoding, i.e., just drop the encoding. It is already tricky when dialect conversion is involved. I'd like to keep the same shape which makes the logic fits MLIR conversion better. I will rename the function, thanks!